[OSDOCS-6135]: Configuring an Azure cluster with short-term creds

jeana-redhat opened the pull request:
Version(s):
4.14+

Issue:
[OSDOCS-6135](https://issues.redhat.com//browse/OSDOCS-6135)

Links to docs preview:
**New content**
* Removed assembly "Manually creating long-term credentials for Azure" (FKA "Manually creating IAM for Azure"), no preview
* [Optional configuration parameters](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installation-config-parameters-azure#installation-configuration-parameters-optional_installation-config-parameters-azure) (new footnote for `credentialsMode` param)
* [Alternatives to storing administrator-level secrets in the kube-system project](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-customizations.html#installing-azure-manual-modes_installing-azure-customizations), sample preview from "Installing a cluster on Azure with customizations". Also included in [network customizations](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-network-customizations#installing-azure-manual-modes_installing-azure-network-customizations), [existing VNet](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-vnet#installing-azure-manual-modes_installing-azure-vnet), and [private cluster](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-private#installing-azure-manual-modes_installing-azure-private).
* Updated links in [Preparing](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/preparing-to-install-on-azure#requirements-for-installing-ocp-on-azure), [UPI](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/installing-azure-user-infra#prerequisites), [passthrough mode](https://64001--docspreview.netlify.app/openshift-enterprise/latest/authentication/managing_cloud_provider_credentials/cco-mode-passthrough#passthrough-mode-permissions-azure), [long-term manual mode](https://64001--docspreview.netlify.app/openshift-enterprise/latest/authentication/managing_cloud_provider_credentials/cco-mode-manual#additional-resources_cco-mode-manual), [short-term manual mode](https://64001--docspreview.netlify.app/openshift-enterprise/latest/authentication/managing_cloud_provider_credentials/cco-short-term-creds#cco-short-term-creds-azure_cco-short-term-creds)
* [Deleting Azure resources with the Cloud Credential Operator utility](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_azure/uninstalling-cluster-azure#cco-ccoctl-deleting-sts-resources_uninstall-cluster-azure)

**Formatting changes for consistent presentation**
* [Creating credentials for OpenShift Container Platform components with the ccoctl tool](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_alibaba/installing-alibaba-default#cco-ccoctl-creating-at-once_installing-alibaba-default) (AliCloud, includes updated cred req list for 4.14)
* [Creating AWS resources with the Cloud Credential Operator utility](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-customizations#sts-mode-create-aws-resources-ccoctl_installing-aws-customizations) (AWS, both the "single command" and "individually" paths, includes updated cred req list for 4.14)
* [Deleting AWS resources with the Cloud Credential Operator utility](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_aws/uninstalling-cluster-aws#cco-ccoctl-deleting-sts-resources_uninstall-cluster-aws)
* [Creating GCP resources with the Cloud Credential Operator utility](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-gcp-customizations#cco-ccoctl-creating-at-once_installing-gcp-customizations) (GCP, includes updated cred req list for 4.14)
* [Deleting GCP resources with the Cloud Credential Operator utility](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/uninstalling-cluster-gcp#cco-ccoctl-deleting-sts-resources_uninstalling-cluster-gcp)
* Optional configuration parameters ([AWS](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-aws-customizations#installation-configuration-parameters-optional_installing-aws-customizations), [GCP](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installation-config-parameters-gcp#installation-configuration-parameters-optional_installation-config-parameters-gcp); moved admonitions into table footnote)
* [Configuring IAM for Nutanix](https://64001--docspreview.netlify.app/openshift-enterprise/latest/installing/installing_nutanix/installing-nutanix-installer-provisioned#manually-create-iam-nutanix_installing-nutanix-installer-provisioned)
* [Updating cloud provider resources with the Cloud Credential Operator utility](https://64001--docspreview.netlify.app/openshift-enterprise/latest/updating/preparing_for_updates/preparing-manual-creds-update#cco-ccoctl-upgrading_preparing-manual-creds-update)
* [Updating cloud provider resources with manually maintained credentials](https://64001--docspreview.netlify.app/openshift-enterprise/latest/updating/preparing_for_updates/preparing-manual-creds-update#manually-maintained-credentials-upgrade_preparing-manual-creds-update)

QE review:
- [x] QE has [approved](https://redhat-internal.slack.com/archives/C04TMSTHUHK/p1697533670080299?thread_ts=1696849431.278319&cid=C04TMSTHUHK) this change.

Additional information:

jeana-redhat commented on modules/cco-ccoctl-creating-at-once.adoc:
```diff
@@ -127,122 +156,118 @@ endif::alibabacloud-default,alibabacloud-customizations,alibabacloud-vpc[]
 ----
-+
 <1> The Cloud Controller Manager Operator CR is required.
+<2> The Cluster API Operator CR is #required#.
```
Per Andrew during standup today, CAPI won't work with AD Workload ID and should be documented accordingly here.

jeana-redhat replied:
This should be irrelevant due to the new `--included` flag. Leaving this PR alone because that work is in #62148

jstuever commented:
/cc

jstuever commented on modules/cco-ccoctl-creating-at-once.adoc:
```diff
@@ -380,13 +427,16 @@ openshift-image-registry-installer-cloud-credentials-credentials.yaml
 openshift-machine-api-aws-cloud-credentials-credentials.yaml
 ----
-//Would love a GCP version of the above output.
+//Would love a GCP and Azure version of the above output.
```
azure-ad-pod-identity-webhook-config.yaml
cluster-authentication-02-config.yaml
openshift-cloud-controller-manager-azure-cloud-credentials-credentials.yaml
openshift-cloud-network-config-controller-cloud-credentials-credentials.yaml
openshift-cluster-api-capz-manager-bootstrap-credentials-credentials.yaml
openshift-cluster-csi-drivers-azure-disk-credentials-credentials.yaml
openshift-cluster-csi-drivers-azure-file-credentials-credentials.yaml
openshift-image-registry-installer-cloud-credentials-credentials.yaml
openshift-ingress-operator-cloud-credentials-credentials.yaml
openshift-machine-api-azure-cloud-credentials-credentials.yaml

jeana-redhat commented on authentication/managing_cloud_provider_credentials/cco-mode-manual.adoc:
```diff
@@ -20,7 +20,7 @@ For information about configuring your cloud provider to use manual mode, see th
 == Manual mode with cloud credentials created and managed outside of the cluster
 
-An AWS or GCP cluster that uses manual mode might be configured to create and manage cloud credentials from outside of the cluster using the AWS Security Token Service (STS) or GCP Workload Identity. With this configuration, the CCO uses short-term credentials for different components.
+AWS, GCP, and global Azure clusters that uses manual mode might be configured to create and manage cloud credentials from outside of the cluster. This credentials management strategy uses the AWS Security Token Service (STS), GCP Workload Identity, or Azure AD Workload Identity. In this configuration, the CCO uses temporary credentials for different components.
```
For the clash, I will defer to the formatting in #64848 - I think having this as a note makes more sense than a subheading since the page it's on is for long-term manual creds. Pushing a commit with new wording, lmk what you think. If it works here, I'll put it in the other PR as well :)

jeana-redhat commented:
Squashed all previous commits. This establishes the "old way" without the new `--included` method. Since Azure short-term creds and that new param both GA in 4.14, I am updating this to use the new param. Will do so in a separate commit so that comparison is easy (if needed)

jstuever commented:
lgtm

jeana-redhat commented on modules/cco-ccoctl-deleting-sts-resources.adoc:
```diff
@@ -2,52 +2,82 @@
+
+. Delete the {cp} resources that `ccoctl` created by running the following command:
+endif::gcp-workload-id[]
+ifdef::aws-sts,azure-workload-id[]
+* Delete the {cp} resources that `ccoctl` created by running the following command:
+endif::aws-sts,azure-workload-id[]
```
Azure requires a couple other parameters, see note here
https://github.com/openshift/openshift-docs/pull/62148#issuecomment-1761055077

jeana-redhat replied:
Added in https://github.com/openshift/openshift-docs/pull/64001/commits/8b47f7af10829104ac6569615d52c280dbcd4240

jeana-redhat commented on installing/installing_azure_stack_hub/installing-azure-stack-hub-user-infra.adoc:
```diff
@@ -83,7 +83,7 @@ include::modules/installation-user-infra-generate-k8s-manifest-ignition.adoc[lev
 [role="_additional-resources"]
 .Additional resources
-* xref:../../installing/installing_azure/manually-creating-iam-azure.adoc#manually-create-iam_manually-creating-iam-azure[Manually creating long-term credentials]
+* xref:../../installing/installing_azure/installing-azure-customizations.adoc#manually-create-iam_installing-azure-customizations[Manually creating long-term credentials]
```
Replaced a defunct link, but not entirely sure this should be here. Asked [here](https://github.com/openshift/openshift-docs/pull/64288/files#r1358695929).

huangmingxia commented:
Hi @jeana-redhat 
One thing I'm not sure about, regarding Azure Government (MAG) and Azure stack hub Cloud, we currently have not delivered it. Whether it needs to be marked in the document. Our slack channel: https://redhat-internal.slack.com/archives/C04TMSTHUHK/p1696849431278319 
Others are great to me.

jeana-redhat commented:
@huangmingxia Thanks for that! I'll catch up in that channel and make sure to only include the versions we are confident about supporting :smiley:

jeana-redhat commented:
Ok @huangmingxia I think this is all correct now with your feedback from above. 
* For MAG, I reverted the content to its existing state, and update the "manually creating IAM" link to point to "manually creating long term credentials" (new title, new location, same process).
* For Stack Hub, I am not including any workload ID content, though I have updated a link in that workflow as well
* Not in this PR, but work already merged breaks Stack Hub out so that it is clear that only long-term manual credentials are supported (see the table in [this section](https://64001--docspreview.netlify.app/openshift-enterprise/latest/authentication/managing_cloud_provider_credentials/about-cloud-credential-operator#about-cloud-credential-operator-modes_about-cloud-credential-operator))

Does that all seem right to you?

jeana-redhat commented:
Squashed commits for peer review

michaelryanpeter commented on modules/installation-configuration-parameters.adoc:
```diff
@@ -656,6 +648,23 @@ ifdef::ibm-power-vs[]
+1. Not all CCO modes are supported for all cloud providers. For more information about CCO modes, see the "Managing cloud provider credentials" entry in the _Authentication and authorization_ content.
++
+ifdef::aws,gcp[]
+[NOTE]
```
I am calling out that we try to avoid stacked admonitons, but looking at this table, and the surrounding content, I am not sure what to suggest as a better alternative.

jeana-redhat replied:
Yeah, I don't know what the eventual solution to these tables will be, but I suspect this workaround I have will not scale because of that consideration. At least a few people are trying to get a better solution to this situation, so hopefully we'll have something slick soon :crossed_fingers:

jeana-redhat commented:
rebase in preparation for merge

jeana-redhat commented:
/cherrypick enterprise-4.14
